### DISCUSSION TITLE
The last post

### DISCUSSION THREAD

**Discussion by TANSixu** [SUSTech-CS305-Fall22/CS305-Project-Skeleton#83] [Announcements]
# Remote containers will be shut down by Feb.19 23:55
Please download/synchronize your code if you would like to keep them.
I was really thankful for our server because it did not crash during our presentation :)

# Excellent implementation from our students
The champion and runner-up have agreed to make their code public for reference. You are encouraged to try their code and learn from their implementation.

## Champion
**Team 19**: Wenqian Yan @YanWQ-monad, Yuetong Huang, Zhe Ding
Their code can be found [here](https://gist.github.com/YanWQ-monad/108066d2b93c2cc671d5b611acdbd600)
Their congestion control algorithm is based on [BBR](https://cloud.google.com/blog/products/networking/tcp-bbr-congestion-control-comes-to-gcp-your-internet-just-got-faster), which helps their implementation **significantly** outperform that of other groups.
Congrats to them and thanks to them for kindly sharing their code!

## Runner-up
**Team 26**: Qizhi Chen, Baiqin Deng, Qiujiang Chen
Their code can be provided upon request. Please reach out to me if you are interested.
The highlight of their code is a well-implemented SACK mechanism.
Congrats to them and thanks to them for kindly sharing their code!

# More resources about networks
## Content review
If you are going to take your final exam at the beginning of SP23, you may find this [textbook website](https://media.pearsoncmg.com/ph/esm/ecs_kurose_compnetwork_8/cw/) helpful. It provides concepts check questions, animations, as well as exercises.

## Extension
1. [CS144 Introduction to Computer Networking](https://cs144.github.io/) from Stanford. It provides a more comprehensive introduction to networks.
2. [CS自学指南](https://csdiy.wiki/) A list of all good CS courses around the world. Strongly recommended!

# 有缘再见
Sincere thanks to every one of you. Wish all of you the best of luck in your future study!

**Comment by Leo-Adventure** [comment]
https://github.com/Leo-Adventure/CS305-Computer-Network/blob/main/计算机网络：自顶向下方法%20概念整理.pdf

希望做的计网概念整理也能够帮助到大家一些，考试加油~

**Reply by TANSixu** [reply]
wow！感谢分享！
